test/data/javascript.js

```diff
@@ -92,6 +94,800 @@ exports.test_data = {
+                    '    b: bval,',
+                    '    c: cval',
+                    '};',
+                    'var d = e ? f : g;'
+                ]
+            }, {
+                comment: 'catch-all, includes brackets and other various code',
+                input: inputlib.operator_position.catch_all,
+                output: [
+                    'var d = 1;',
+                    'if (a === b && c) {',
+                    '    d = (c * everything / something_else) %',
+                    '        b;',
+                    '    e',
+                    '        += d;',
+                    '',
+                    '} else if (!(complex && simple) ||',
+                    '    (emotion && emotion.name === "happy")) {',
+                    '    cryTearsOfJoy(many ||',
+                    '        anOcean ||',
+                    '        aRiver);',
+                    '}'
+                ]
+            }
+        ]
+    }, {
+        name: "operator_position option - set to default (undefined)",
+        options: [
+            { name: "operator_position", value: "undefined" },
+            { name: "preserve_newlines", value: "false" },
+        ],
+        tests: [
+            {
+                comment: 'comprehensive, but simple',
+                unchanged: inputlib.operator_position.comprehensive_simple
+            }, {
+                comment: 'comprehensive, various newlines',
+                input: inputlib.operator_position.comprehensive_newlines,
+                output: [
+                    'var res = a + b - c / d * e % f;',
+                    'var res = g & h | i ^ j;',
+                    'var res = (k && l || m) ? n : o;',
+                    'var res = p >> q << r >>> s;',
+                    'var res = t === u !== v != w == x >= y <= z > aa < bb;'
+                ]
+            }, {
+                comment: 'catch-all, includes brackets and other various code',
+                input: inputlib.operator_position.colon_special_case,
+                output: [
+                    'var a = {',
+                    '    b: bval,',
+                    '    c: cval',
+                    '};',
+                    'var d = e ? f : g;'
+                ]
+            }, {
+                comment: 'handle colon special case',
+                input: inputlib.operator_position.catch_all,
+                output: [
+                    'var d = 1;',
+                    'if (a === b && c) {',
+                    '    d = (c * everything / something_else) % b;',
+                    '    e += d;',
+                    '} else if (!(complex && simple) || (emotion && emotion.name === "happy")) {',
+                    '    cryTearsOfJoy(many || anOcean || aRiver);',
+                    '}'
+                ]
+            }
+        ]
+    }, {
+        name: "operator_position option - set to default (undefined)",
+        options: [
+            { name: "operator_position", value: "undefined" },
```

Comment:
You know, this would be one spot where the matrix structure would help you.  You could clearly show that your testing for `operator_position = undefined` to have the same behavior as the default value.